opm/input/eclipse/EclipseState/Grid/FieldPropsManager.cpp

```diff
@@ -174,6 +179,37 @@ FieldPropsManager::getTran() const
     return this->fp->getTran();
 }
 
+void apply_action(const Fieldprops::ScalarOperation& op,
+                  const Fieldprops::FieldData<double>& action_data,
+                  std::size_t index,
+                  std::vector<double>& data)
+{
+    switch (op) {
+    case Fieldprops::ScalarOperation::EQUAL:
+        data[index] = action_data.data[index];
+        break;
+
+    case Fieldprops::ScalarOperation::MUL:
+        data[index] *= action_data.data[index];
+        break;
+
+    case Fieldprops::ScalarOperation::ADD:
+        data[index] += action_data.data[index];
+        break;
+
+    case Fieldprops::ScalarOperation::MAX:
+        data[index] = std::min(action_data.data[index], data[index]);
```

Comment:
Thanks for the explanation! If the variable compared to had been named upper_bound it might have been easier to understand, but the best choice here is to add an explanatory comment, for example:
```
// The MAX operator sets an upper bound, the
// resulting value is therefore the minimum of the
// input value and that upper bound.
```